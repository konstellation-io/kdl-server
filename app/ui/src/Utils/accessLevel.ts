import { AccessLevel } from 'Graphql/types/globalTypes';

export const mapProjectAccessLevel = {
  [AccessLevel.ADMIN]: 'Owner',
  [AccessLevel.MANAGER]: 'Developer',
  [AccessLevel.VIEWER]: 'Member',
}
