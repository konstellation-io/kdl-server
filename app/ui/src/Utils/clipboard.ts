import { toast } from 'react-toastify';

export function copyToClipboard(value: string) {
  const tempInput = document.createElement('input');
  tempInput.value = value;
  document.body.appendChild(tempInput);
  tempInput.select();
  document.execCommand('copy');
  document.body.removeChild(tempInput);
  toast.info('Copied to clipboard');
}

export async function copyAndToast(value: string) {
  await navigator.clipboard.writeText(value);

  toast.info('Copied to clipboard');
  toast.clearWaitingQueue();
}
