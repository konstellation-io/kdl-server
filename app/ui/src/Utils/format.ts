import moment from 'moment-timezone';

export function formatDate(date: Date, showTime = false) {
  const format = showTime ? 'lll' : 'll';
  return moment(date).format(format);
}
