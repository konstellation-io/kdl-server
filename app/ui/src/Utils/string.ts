const ID_MAX_LENGTH = 20;
const INVALID_ID_CHARS_REGEXP = /[^a-z0-9-]/g;
const SPACES_REGEXP = / +/g;

export function generateSlug(text: string) {
  return text
    .trim()
    .toLowerCase()
    .replace(SPACES_REGEXP, '-')
    .replace(INVALID_ID_CHARS_REGEXP, '')
    .substr(0, ID_MAX_LENGTH);
}
export function replaceAll(text: string, regex: RegExp, replaceWith: string): string {
  return text.replace(new RegExp(regex, 'g'), replaceWith);
}

export function getErrorMsg(validation: string | boolean): string {
  return validation === true ? '' : (validation as string);
}
