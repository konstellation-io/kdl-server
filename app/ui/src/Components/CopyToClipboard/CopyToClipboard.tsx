import React, { FC, MouseEvent } from 'react';

import { Button } from 'kwc';
import IconCopy from '@material-ui/icons/FileCopyOutlined';
import { copyAndToast } from 'Utils/clipboard';
import cx from 'classnames';
import styles from './CopyToClipboard.module.scss';

type Props = {
  children: string;
  className?: string;
};
const CopyToClipboard: FC<Props> = ({ children, className }) => {
  function onCopy(event?: MouseEvent<HTMLDivElement>) {
    event?.stopPropagation();
    event?.preventDefault();

    copyAndToast(children);
  }

  return <Button label="" Icon={IconCopy} onClick={onCopy} className={cx(styles.container, className)} />;
};

export default CopyToClipboard;
