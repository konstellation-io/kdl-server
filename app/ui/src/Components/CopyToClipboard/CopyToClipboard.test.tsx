import * as clipboard from 'Utils/clipboard';

import CopyToClipboard from './CopyToClipboard';
import * as React from 'react';
import { shallow } from 'enzyme';

let component;
const textToCopy = 'foo';

beforeEach(() => {
  component = shallow(<CopyToClipboard>{textToCopy}</CopyToClipboard>);
});
describe('CopyToClipboard component', () => {
  test('Component match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('copy text on click', () => {
    // Arrange.
    const mockCopyAndToast = jest.fn();
    clipboard.copyAndToast = mockCopyAndToast;

    // Act.
    component.simulate('click');

    // Assert.
    expect(mockCopyAndToast).toHaveBeenCalledTimes(1);
    expect(mockCopyAndToast).toHaveBeenCalledWith(textToCopy);
  });
});
