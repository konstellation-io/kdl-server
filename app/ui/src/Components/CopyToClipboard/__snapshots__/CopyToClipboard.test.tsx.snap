// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipboard component Component match snapshot 1`] = `
<P
  Icon={
    Object {
      "$$typeof": Symbol(react.memo),
      "compare": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    }
  }
  className="container"
  label=""
  onClick={[Function]}
/>
`;
